### PR TITLE
dwdiff: init at 2.1.4

### PR DESCRIPTION
**Pull request by onny**
###### Description of changes

Fixes package request https://github.com/NixOS/nixpkgs/issues/185013

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/dwdiff/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [
+    gettext
+    icu
+  ];
+
+  meta = with lib; {
+    description = "An application to split, merge, rotate and mix PDF files";
```
Comment (unhammer) [inline pkgs/applications/misc/dwdiff/default.nix:26]: maybe `"a word-level diff with color support which lets you specify delimiter- and whitespace-characters"` ?

File: pkgs/applications/misc/dwdiff/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    homepage = "https://gitlab.com/scarpetta/pdfmixtool";
```
Comment (unhammer) [inline pkgs/applications/misc/dwdiff/default.nix:27]: https://os.ghalkes.nl/dwdiff.html

File: pkgs/applications/editors/jetbrains/versions.json [hunk 3]
```diff
@@ -84,10 +84,10 @@
     "rider": {
       "update-channel": "Rider RELEASE",
       "url-template": "https://download.jetbrains.com/rider/JetBrains.Rider-{version}.tar.gz",
-      "version": "2022.1.2",
```
Comment (unhammer) [inline pkgs/applications/editors/jetbrains/versions.json:87]: unrelated change?

**Review by unhammer — CHANGES_REQUESTED** [review]

**Comment by onny** [comment]
@unhammer Thank you for reviewing. Sorry that this PR was quite messy.

File: pkgs/applications/misc/dwdiff/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [
+    gettext
+    icu
+  ];
+
+  meta = with lib; {
+    description = "A front-end for the diff program that operates at the word level instead of the line level";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/dwdiff/default.nix:26]:
```suggestion
    description = "Front-end for the wdiff program that operates at the word level instead of the line level";
```

File: pkgs/applications/misc/dwdiff/default.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+, pkg-config
+, icu
+}:
+
+stdenv.mkDerivation rec {
+  pname = "dwdiff";
+  version = "2.1.4";
+
+  src = fetchurl {
+    url = "https://os.ghalkes.nl/dist/${pname}-${version}.tar.bz2";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/dwdiff/default.nix:14]:
```suggestion
    url = "https://os.ghalkes.nl/dist/dwdiff-${version}.tar.bz2";
```

File: pkgs/applications/misc/dwdiff/default.nix (hunk 4 above)
Comment (unhammer) [reply pkgs/applications/misc/dwdiff/default.nix:26]: That doesn't make sense, [wdiff](https://www.gnu.org/software/wdiff/) is a separate front-end for diff that operates at the word level
